authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,1015 @@
+    } catch (RangerServiceException e) {
+      // Ignore exception to support idempotent operation
+    }
+    return Boolean.TRUE;
+  }
+
+  @Override
+  public Boolean onRoleUpdated(Role role, RoleChange... changes) throws RuntimeException {
+    for (RoleChange change : changes) {
+      boolean execResult;
+      if (change instanceof RoleChange.AddSecurableObject) {
+        execResult = doAddSecurableObject((RoleChange.AddSecurableObject) change);
+      } else if (change instanceof RoleChange.RemoveSecurableObject) {
+        execResult =
+            doRemoveSecurableObject(role.name(), (RoleChange.RemoveSecurableObject) change);
+      } else if (change instanceof RoleChange.UpdateSecurableObject) {
+        execResult =
+            doUpdateSecurableObject(role.name(), (RoleChange.UpdateSecurableObject) change);
+      } else {
+        throw new IllegalArgumentException(
+            "Unsupported role change type: "
+                + (change == null ? "null" : change.getClass().getSimpleName()));
+      }
+      if (!execResult) {
+        return Boolean.FALSE;
+      }
+    }
+
+    return Boolean.TRUE;
+  }
+
+  /**
+   * Set or transfer the ownership of the metadata object. <br>
+   *
+   * @param metadataObject The metadata object to set the owner.
+   * @param preOwner The previous owner of the metadata object. If the metadata object doesn't have
+   *     owner, then the preOwner will be null and newOwner will be not null.
+   * @param newOwner The new owner of the metadata object. If the metadata object already have
+   *     owner, then the preOwner and newOwner will not be null.
+   */
+  @Override
+  public Boolean onOwnerSet(MetadataObject metadataObject, Owner preOwner, Owner newOwner)
+      throws RuntimeException {
+    // 1. Set the owner of the metadata object
+    // 2. Transfer the ownership from preOwner to newOwner of the metadata object
+    check(newOwner != null, "The newOwner must be not null");
+
+    RangerPolicy policy = findManagedPolicy(metadataObject);
+    if (policy != null) {
+      policy.getPolicyItems().stream()
+          .filter(
+              policyItem -> {
+                return policyItem.getAccesses().stream()
+                    .allMatch(
+                        policyItemAccess -> {
+                          return ownerPrivileges.contains(policyItemAccess.getType());
+                        });
+              })
+          .forEach(
+              policyItem -> {
+                if (preOwner != null) {
+                  if (preOwner.type() == Owner.Type.USER) {
+                    policyItem.getUsers().removeIf(preOwner.name()::equals);
+                  } else {
+                    policyItem.getGroups().removeIf(preOwner.name()::equals);
+                  }
+                }
+                if (newOwner != null) {
+                  if (newOwner.type() == Owner.Type.USER) {
+                    policyItem.getUsers().add(newOwner.name());
```

Comment:
Should we verify whether the user exists in Ranger?

Reply:
Yes, you are right.
I added verify whether the user exists in Ranger.